source/common/event/dispatcher_impl.h

```diff
@@ -93,7 +93,7 @@ class DispatcherImpl : Logger::Loggable<Logger::Id::main>,
   // Validate that an operation is thread safe, i.e. it's invoked on the same thread that the
   // dispatcher run loop is executing on. We allow run_tid_ to be empty for tests where we don't
   // invoke run().
-  bool isThreadSafe() const {
+  bool isThreadSafe() const override {
```

Comment:
this was only part of DispatcherImpl and only used there before this PR, Can we add some explicit test to this? esp. for the cases it should return false.